### PR TITLE
add LockPostgreSQL and BenchmarkDistLockers

### PR DESCRIPTION
**Pull request by leeym**
Add the following two

- `LockPostgreSQL` implements `DistLocker` using [advisory locks in PostgreSQL](https://www.postgresql.org/docs/current/explicit-locking.html#ADVISORY-LOCKS)
- `BenchmarkDistLockers` to compare the performance for distributed lockers

```
goarch: amd64
pkg: github.com/mennanov/limiters
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkDistLockers
2024/03/16 15:44:22 Connected to 127.0.0.1:2181
2024/03/16 15:44:22 authenticated: id=72064875209293827, timeout=4000
2024/03/16 15:44:22 re-submitting `0` credentials after reconnect
BenchmarkDistLockers/LockEtcd
BenchmarkDistLockers/LockEtcd-12                     171           6576282 ns/op
BenchmarkDistLockers/LockConsul
BenchmarkDistLockers/LockConsul-12                   100          15450609 ns/op
BenchmarkDistLockers/LockZookeeper
BenchmarkDistLockers/LockZookeeper-12                265           4537545 ns/op
BenchmarkDistLockers/LockRedis
BenchmarkDistLockers/LockRedis-12                   1238            984753 ns/op
BenchmarkDistLockers/LockMemcached
BenchmarkDistLockers/LockMemcached-12                816           1608760 ns/op
BenchmarkDistLockers/LockPostgreSQL
BenchmarkDistLockers/LockPostgreSQL-12               670           1664998 ns/op
```

It will be easier to review in [unified view and hide whitespace](https://github.com/mennanov/limiters/pull/39/files?diff=unified&w=1).

I should reopen the previous PR instead of creating a new one. Anyway, the difference is that in https://github.com/mennanov/limiters/pull/38 I used session-level advisory lock. Since in the test we share the same database connection, thus the same session, between two lockers, they interfere with each other as the lock acquired by one locker can be released by the other one. In this PR I use transaction-level advisory lock, so each locker has its own transaction even if they share the same database connection.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,4 +2,4 @@ all: test
 
 test:
 	docker-compose up -d
-	ETCD_ENDPOINTS="127.0.0.1:2379" REDIS_ADDR="127.0.0.1:6379" ZOOKEEPER_ENDPOINTS="127.0.0.1" CONSUL_ADDR="127.0.0.1:8500" AWS_ADDR="127.0.0.1:8000" MEMCACHED_ADDR="127.0.0.1:11211" go test -race -v -failfast
+	ETCD_ENDPOINTS="127.0.0.1:2379" REDIS_ADDR="127.0.0.1:6379" ZOOKEEPER_ENDPOINTS="127.0.0.1" CONSUL_ADDR="127.0.0.1:8500" AWS_ADDR="127.0.0.1:8000" MEMCACHED_ADDR="127.0.0.1:11211" go test -race -v -failfast -bench=.
```
Comment (mennanov) [inline Makefile:5]: Do you need to add the `POSTGRES_ADDR` here?

**Review by mennanov — CHANGES_REQUESTED** [review]
Looks good!
I left just 1 small comment.

Thanks!

File: Makefile (hunk 1 above)
Comment (leeym) [reply Makefile:5]: Sure, currently I just hardcode as localhost:5432. Let me add that to make it configurable.

Comment (leeym) [reply Makefile]: I added `POSTGRES_URL`

**Review by mennanov — APPROVED** [review]

**Pull request merged**